templates/_helpers.tpl

```diff
@@ -58,8 +60,10 @@ Selector labels
 {{- define "common.selectorLabels" -}}
 app.kubernetes.io/name: {{ include "common.name" . }}
 app.kubernetes.io/instance: {{ .root.Release.Name }}
-{{- if .service.serviceName }}
-app.kubernetes.io/component: {{ .service.serviceName }}
+{{- if hasKey .service "serviceName" }}
+app.kubernetes.io/component: {{ printf "%s" .service.serviceName }}
+{{- else if hasKey . "serviceName" }}
+app.kubernetes.io/component: {{ printf "%s" .serviceName }}
```

Comment:
Same order as above:
```suggestion
{{- if hasKey . "serviceName" }}
app.kubernetes.io/component: {{ printf "%s" .serviceName }}
{{- else if hasKey .service "serviceName" }}
app.kubernetes.io/component: {{ printf "%s" .service.serviceName }}
```

Reply:
no: we want a precedence on what is defined within the service.serviceName before taking the default value. We discussed this yesterday